Source/WebCore/html/HTMLAttachmentElement.cpp

```diff
@@ -234,20 +270,22 @@ class AttachmentSaveEventListener final : public EventListener {
 
 void HTMLAttachmentElement::updateSaveButton(bool show)
 {
-    if (!m_containerElement)
-        return;
-
     if (!show) {
         if (m_saveButton) {
```

Comment:
Any reason this can't just be `if (m_saveArea)`? Looks like `m_saveArea` and `m_saveButton` have the same lifetime?

Do we need both?

Reply:
Yes they have the same lifetimes.
But I believe we do need both: `m_saveArea` contains `m_saveButton` but they have different sizes; we need `m_saveButton` in code elsewhere to find its location in the attachment; and we need `m_saveArea` as the top element to be removed when we don't need the save button anymore (otherwise it would still take some space).

Reply:
I expect we could achieve these requirements by using `display: flex` rather than `display: grid` in some places, but that would be a bigger change, and the current approach is also fine.